MyApplication/app/src/main/java/com/scottlindley/listview_listadapters_lab/MainActivity.java

```diff
@@ -0,0 +1,154 @@
+package com.scottlindley.listview_listadapters_lab;
+
+import android.graphics.Color;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+public class MainActivity extends AppCompatActivity {
+
+    private ListView mListView;
+    private BaseAdapter mBaseAdapter;
+    private ArrayList<Book> mBooks;
+    private Button mSortButton;
+    private Spinner mSpinner;
+    private ArrayList<String> mSortingOptions;
+    private ArrayAdapter<String> mArrayAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mSortButton = (Button)findViewById(R.id.sort_button);
+
+        initializeBooksArray();
+
+        initializeSpinner();
+
+        initializeListView();
+
+        setListViewClick();
+
+        setSortButtonClick();
```

Comment:
I like how you keep `onCreate()` very clean and readable by using helper methods. Makes grading easy!